### PR TITLE
fix:sideBar updating URL

### PR DESCRIPTION
**Pull request by AnoukRImola**
## Changelog

- The `use-sidebar` hook was modified to allow updates based on the URL.
- Additionally, I fixed some minor warnings in a few Tailwind styles.

https://github.com/user-attachments/assets/b20994bc-b8f3-45c5-942d-26eeb11b27c9

## Summary by Sourcery

Update the sidebar functionality to dynamically adjust based on the URL path, ensuring the correct category and chatbot are selected. Additionally, refine Tailwind CSS styles to address minor warnings and enhance code consistency.

Bug Fixes:
- Fix the sidebar to update based on the URL path, ensuring correct category and chatbot selection.

Enhancements:
- Modify Tailwind CSS styles to remove minor warnings and improve code consistency.

## Summary by CodeRabbit

- **New Features**
	- Enhanced sidebar functionality with improved routing responsiveness and state management.
	- Improved management of expanded categories in the sidebar based on URL changes.

- **Style**
	- Simplified CSS class names for better readability and consistency.
	- Updated translation properties in the chatbot details component for improved layout.
	- Adjusted styling for the close icon within the sheet component.

- **Bug Fixes**
	- Improved sidebar state management by storing it in local storage.

### CODE REVIEW

**Review by AndlerRL — CHANGES_REQUESTED** [review]
Looking good! I see the updates on the sidebar, I noticed the category to have the active state however, I also noticed that the category domain is not opening his bot options and therefore highlighting the selected bot that matched with the pathname is not happening (as per original design). Please make that update and it will be good to go! :+1:

**Comment by AnoukRImola** [comment]
> Looking good! I see the updates on the sidebar, I noticed the category to have the active state however, I also noticed that the category domain is not opening his bot options and therefore highlighting the selected bot that matched with the pathname is not happening (as per original design). Please make that update and it will be good to go! 👍

**Comment by AnoukRImola** [comment]
> Looking good! I see the updates on the sidebar, I noticed the category to have the active state however, I also noticed that the category domain is not opening his bot options and therefore highlighting the selected bot that matched with the pathname is not happening (as per original design). Please make that update and it will be good to go! 👍

Ready :)

File: apps/masterbots.ai/components/layout/sidebar/sidebar-link.tsx
Comment (AndlerRL) [inline apps/masterbots.ai/components/layout/sidebar/sidebar-link.tsx:1]: I see the parent element for the sidebar-link is [a div](https://github.com/bitcashorg/masterbots/pull/286/files#diff-a0581528eb5de30db003d237b7699c2bd10284608a3e1e3895df354f4567d58eR64). Please use a Link component for chat navigation and keep the div for browse sidebar. That way we can update and hear the category whenever the path updates

File: apps/masterbots.ai/components/layout/sidebar/sidebar-link.tsx [hunk 1]
```diff
@@ -26,16 +26,22 @@ export default function SidebarLink({ category, isFilterMode }: SidebarLinkProps
...
+  const isExpanded = expandedCategories.includes(category.categoryId)
 
   const handleClickCategory = useCallback((e: React.MouseEvent) => {
     e.stopPropagation()
     if (!isFilterMode) {
-      setIsExpanded(prev => !prev)
-      setActiveCategory(prev => prev === category.categoryId ? null : category.categoryId)
-    }
-  }, [category.categoryId, setActiveCategory, isFilterMode])
+      setExpandedCategories(prev =>
```
Comment (AndlerRL) [inline apps/masterbots.ai/components/layout/sidebar/sidebar-link.tsx:37]: I am looking that in here we are expanding the active category and leaving the previous one open, like for the filter when we click them, they open all and that is fine for that scenario. Chat scenario for navigation, the active category should toggle the expanded category depending which one is active (as it was before).

If we have an active bot, that bot category domain should remain active and when the user selects another category, the previous closes and updates the path to the selected category.

**Review by AndlerRL — CHANGES_REQUESTED** [review]
Thanks Anouk, I would have to request additional changes after the latest update. I added the comments related to the feedback. I am sharing the video showing the issue.

https://github.com/user-attachments/assets/4b065220-f6a0-442e-8096-95659f10cd5c

> TL;DR: The sidebar navigation is not OK. Sidebar navigation has a regression with the path update. Each previous selected category should close after selecting another one.

**Review by AndlerRL — APPROVED** [review]

**Pull request merged**